### PR TITLE
script to generate png files from xyz coordinates with mayavi

### PR DESCRIPTION
**Pull request by burakbayramli**
The result on dam break can be seen here

### CODE REVIEW

File: scripts/visualize_xyz_mayavi.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+fig = mlab.figure(figure=None, fgcolor=(0., 0., 0.), bgcolor=(1, 1, 1), engine=None)
+for i,f in enumerate(files):
+    print (f)
+    df = pd.read_csv(f,sep=' ',header=None)
+    df = np.array(df)
+
+    r = np.ones(len(df))*0.03
+
+    color=(0.2, 0.4, 0.5)
```
Comment (doyubkim) [inline scripts/visualize_xyz_mayavi.py:23]: Make it as a parameter?

File: scripts/visualize_xyz_mayavi.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# the png files can be combined with ImageMagick,
+# convert -delay 30 /tmp/simsph/*.png $HOME/Downloads/simsph.gif
+import pandas as pd, os, sys
+import numpy as np, glob
+from mayavi import mlab
```
Comment (doyubkim) [inline scripts/visualize_xyz_mayavi.py:5]: Let's add a comment in the header part of this file saying that this script depends on `mayavi`.

**Review by doyubkim — CHANGES_REQUESTED** [review]
Thanks for such a useful script! I made some comments, mostly around the input parameters.

File: scripts/visualize_xyz_mayavi.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+mlab.options.offscreen = True
+BS = 3.0
```
[the remark this answers is not in the document]
Comment (burakbayramli) [reply scripts/visualize_xyz_mayavi.py:8]: Hi, BS is to specify the length of each side of the 3D cube. I should probably rename it to something clearer

File: scripts/visualize_xyz_mayavi.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+if (os.path.exists("/tmp/simsph") == False):
```
[the remark this answers is not in the document]
Comment (burakbayramli) [reply scripts/visualize_xyz_mayavi.py:9]: No problem

[the remark this answers is not in the document]
Comment (burakbayramli) [reply scripts/visualize_xyz_mayavi.py:9]: I'll make it user input

File: scripts/visualize_xyz_mayavi.py (hunk 2 above)
Comment (burakbayramli) [reply scripts/visualize_xyz_mayavi.py:5]: Will do

File: scripts/visualize_xyz_mayavi.py (hunk 1 above)
Comment (burakbayramli) [reply scripts/visualize_xyz_mayavi.py:27]: id leave this to user of this script, they can tinker with colors at code level

File: scripts/visualize_xyz_mayavi.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# the png files can be combined with ImageMagick,
+# convert -delay 30 /tmp/simsph/*.png $HOME/Downloads/simsph.gif
+import pandas as pd, os, sys
```
[the remark this answers is not in the document]
Comment (burakbayramli) [reply scripts/visualize_xyz_mayavi.py]: done

File: scripts/visualize_xyz_mayavi.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    mlab.plot3d([0.0,BS],[0.0,0.0],[BS,BS], color=(0,0,0), tube_radius=None, figure=fig)
+    mlab.plot3d([BS,BS],[0.0,BS/2],[BS,BS], color=(0,0,0), tube_radius=None, figure=fig)
+    mlab.plot3d([BS,0],[BS/2,BS/2],[BS,BS], color=(0,0,0), tube_radius=None, figure=fig)
+    mlab.plot3d([0,0],[BS/2,BS/2],[BS,0], color=(0,0,0), tube_radius=None, figure=fig)
+    mlab.plot3d([BS,BS],[0.0,0.0],[0.0,BS], color=(0,0,0), tube_radius=None, figure=fig)
+    mlab.plot3d([BS,BS],[0.0,BS/2],[0.0,0.0], color=(0,0,0), tube_radius=None, figure=fig)
+    mlab.plot3d([BS,0.0],[BS/2,BS/2],[0.0,0.0], color=(0,0,0), tube_radius=None, figure=fig)
+    mlab.plot3d([BS,BS],[BS/2,BS/2],[0.0,BS], color=(0,0,0), tube_radius=None, figure=fig)
+
+    mlab.view(azimuth=50, elevation=80, focalpoint=[1, 0.2, 1.1], distance=8.0, figure=fig)
```
[the remark this answers is not in the document]
Comment (burakbayramli) [reply scripts/visualize_xyz_mayavi.py:43]: Id leave this to the user as well

**Comment by burakbayramli** [comment]
done with most changes, plz review

File: scripts/t.sh [hunk 7]
```diff
@@ -0,0 +1 @@
+python visualize_xyz_mayavi.py /home/burak/Downloads/output/fluid-engine-dev/dam_break /tmp/simsph
```
Comment (doyubkim) [inline scripts/t.sh:1]: This file should be removed

**Review by doyubkim — CHANGES_REQUESTED** [review]
Looks good to me other than the test script file! Once it's removed, let's merge it to the main.

File: scripts/t.sh (hunk 7 above)
Comment (burakbayramli) [reply scripts/t.sh]: Done :)

**Review by doyubkim — APPROVED** [review]
Awesome thanks @burakbayramli

**Pull request merged**